### PR TITLE
original dst: allow listener filter overwrite the default setting

### PR DESCRIPTION
**Pull request by wbpcode**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by wbpcode** [comment]
Hmmm

File: envoy/network/filter.h [hunk 1]
```diff
@@ -324,6 +324,14 @@ class ListenerFilterCallbacks {
    * @param success boolean telling whether the filter execution was successful or not.
    */
   virtual void continueFilterChain(bool success) PURE;
+
+  /**
+   * Overwrite the default use original dst setting for current connection. This allow the listener
```
Comment (alyssawilk) [inline envoy/network/filter.h:329]: allow -> allows

File: source/common/listener_manager/active_tcp_socket.h [hunk 2]
```diff
@@ -88,7 +91,7 @@ class ActiveTcpSocket : public Network::ListenerFilterManager,
   // The owner of this ActiveTcpSocket.
   ActiveStreamListenerBase& listener_;
   Network::ConnectionSocketPtr socket_;
-  const bool hand_off_restored_destination_connections_;
+  bool hand_off_restored_destination_connections_{};
```
Comment (alyssawilk) [inline source/common/listener_manager/active_tcp_socket.h:94]: I know this predates you but can you comment? it's not obvious to me what the variable originally did or why or how

**Review by alyssawilk — COMMENTED** [review]
I'm not terribly familiar with the listener filter code - greg may be a better primary reviewer but here's my take :-)
/wait

**Comment by wbpcode** [comment]
@alyssawilk Thanks for the review. they are addressed.

**Comment by wbpcode** [comment]
Hi, @ggreenway, could you take a second look when you have free time? Thanks.

**Review by alyssawilk — APPROVED** [review]

**Pull request merged**